src/middlewares/authentication.js

```diff
@@ -0,0 +1,49 @@
+import { config } from 'dotenv';
+import jwt from 'jsonwebtoken';
+
+import database from '../models';
+import { ApplicationError } from '../helpers/errors';
+
+const { SECRET_KEY } = process.env;
+const { BlacklistedTokens } = database;
+
+config();
+
+export default {
+  /**
+   * Verify Token
+   *
+   * @param {Object} request - the request object
+   *
+   * @return {void} - undefined
+   */
+  verifyToken: async (request) => {
+    const authHeader = request.headers.authorization;
+    if (authHeader === '') throw new ApplicationError(400, 'No token provided. Please signup or login');
+    if (!authHeader) throw new ApplicationError(412, 'Header not set');
```

Comment:
Same as above

Reply:
Also, the error seems too general "Header not set" looks like no header was set, please change it to something like "Authorization header not set"

Reply:
I have made the corrections

Reply:
Done